R/print.data.table.R

```diff
@@ -63,14 +63,23 @@ print.data.table = function(x, topn=getOption("datatable.print.topn"),
     return(invisible(x))
   }
   if (show.indices) {
-    if (is.null(indices(x))) {
-      show.indices = FALSE
-    } else {
-      index_dt <- as.data.table(attributes(attr(x, 'index')))
-      print_names <- paste0("index", if (ncol(index_dt) > 1L) seq_len(ncol(index_dt)) else "", ":", sub("^__", "", names(index_dt)))
-      setnames(index_dt, print_names)
+    indices <- names(attr(x, "index", exact = TRUE))
+    if (length(indices)) {
+      cleaned_indices <- gsub("^__|_", ", ", indices) 
+      cleaned_indices <- sub(", $", "", cleaned_indices) 
```

Comment:
There's probably a way to fix the code above so that it doesn't create a separator at the end of the string in the first place.